src/vivarium/framework/components/manager.py

```diff
@@ -70,19 +89,54 @@ def setup_components(self, builder, configuration):
 
 
 class ComponentInterface:
+    """The component manager interface made available from the builder."""
```

Comment:
We want significantly more detail on the interfaces.  We'll collect the docstrings into the builder concept note to detail what services the builder provides.  The target of the docstrings for the interfaces is users and component developers, so it's not nitty gritty, but should provide at least a heuristic overview of what's happening. and when.